File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.js
function solution(n, m, section) {
    var answer = 0;
    
    let last_start = -1; 
    
    for(start of section){
        if(start > last_start){
            last_start = start + m - 1;
            answer+=1            
        }
    }
        
    
    return answer;
}

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.py
def solution(n, m, section):
    answer = 0
    
    last_painted = -1 #가장 마지막으로 칠한 구간
    
    for start in section: # 칠할 시작 지점
        if start > last_painted:
            last_painted = start + m - 1
            answer+=1
        
    return answer